keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ApplicationController.kt

```diff
@@ -44,32 +44,24 @@ class ApplicationController(
   )
   fun get(
     @PathVariable("application") application: String,
-    @RequestParam("includeDetails", required = false, defaultValue = "false") includeDetails: Boolean
+    @RequestParam("includeDetails", required = false, defaultValue = "false") includeDetails: Boolean,
+    @RequestParam("entities", required = false, defaultValue = "resources") entities: List<String>
   ): Map<String, Any> {
-    if (includeDetails) {
-      var resources = repository.getSummaryByApplication(application)
-      resources = resources.map { summary ->
-        if (actuationPauser.resourceIsPaused(summary.id)) {
-          // we only update the status if the individual resource is paused,
-          // because the application pause is reflected in the response as a top level key.
-          summary.copy(status = PAUSED)
-        } else {
-          summary
+    return mutableMapOf<String, Any>(
+      "applicationPaused" to actuationPauser.applicationIsPaused(application),
+      "hasManagedResources" to applicationService.hasManagedResources(application)
+    ).also { results ->
+      if (includeDetails) {
+        entities.forEach { entity ->
+          results[entity] = when (entity) {
+            "resources" -> applicationService.getResourceSummariesFor(application)
+            "environments" -> applicationService.getEnvironmentSummariesFor(application)
+            "artifacts" -> applicationService.getArtifactSummariesFor(application)
+            else -> throw InvalidRequestException("Unknown entity type: $entity")
+          }
         }
       }
-      val constraintStates = repository.constraintStateFor(application)
-
-      return mapOf(
-        "applicationPaused" to actuationPauser.applicationIsPaused(application),
-        "hasManagedResources" to resources.isNotEmpty(),
-        "resources" to resources,
-        "currentEnvironmentConstraints" to constraintStates
```

Comment:
I removed this per discussions with @erikmunson as constraints will show up elsewhere in the response later.